text/0000-builder-config.md

```diff
@@ -0,0 +1,120 @@
+# Builder Config
+
+## Summary
+
+This RFC proposes an easy way to configure builders to allow specifying a `config.toml` file that allows updating the Buildpack `detect` and `build` environment based on the configuration file.
+
+## Motivation
+
+Often times, especially in enterprise settings, organizations often have to update the buildpacks to use internal mirrors, proxies and other settings which can be configured easily with environment variables.
+
+Some examples include -
+- `GOPROXY`
+- `PIP_INDEX_URL`
+- `npm_config_registry`
+- `BUNDLE_MIRROR__URL`
+
+The buildpack logic in `Paketo` largely remains the same, except these environment variables might need to be injected during the `build` and `detect` phases.
+
+The environment variables may ideally also take precendence over any user provided values to ensure that the operators have full control over their builders.
```

Comment:
```suggestion
The environment variables may be able to take precedence over user-provided values, if configured to, which ensures that operators can take full control over their builders if that is required.
```

I would suggest this because I don't think this should be the default value. I think by default, user-configured values should be the highest precedence. That said, you have the `override` mode, so if set to `override` then an operator could take control of that value.

My rationale for this is a.) we do that in pretty much every other situation, so I would say it's the generally expected behavior b.) I think it would be difficult to debug if something wasn't working right. To know that you're setting an env variable as a user and it's getting overridden by the stack.

I would further suggest that when the mode is set to `override` for a builder env variable we should log a message (probably debug, but possibly info) to indicate this happened.